### PR TITLE
Changes to XDMFEntry to use uint64_t data types

### PR DESCRIPTION
**Pull request by singima**

### CODE REVIEW

File: source/base/data_out_base.cc [hunk 1]
```diff
@@ -7895,7 +7895,7 @@ DataOutInterface<dim, spacedim>::create_xdmf_entry(
   const double                      cur_time,
   const MPI_Comm &                  comm) const
 {
-  unsigned int local_node_cell_count[2], global_node_cell_count[2];
+  std::uint64_t local_node_cell_count[2], global_node_cell_count[2];
```
Comment (tjhei) [inline source/base/data_out_base.cc:7898]: You also have to fix the data type of the MPI call a couple of lines down from here.

**Review by tjhei — COMMENTED** [review]
I think this is good. One might argue to use a different data type and only a 64 bit type if deal.II is compiled with 64 bit and indices, but I think the current approach is easier and handles the case that a large subdivision number pushes the number of nodes above 32bits.

File: source/base/data_out_base.cc (hunk 1 above)
Comment (singima) [reply source/base/data_out_base.cc:7898]: Thank you, good catch.  Should be fixed now!

**Comment by tjhei** [comment]
Do the xdmf files still generate correctly? Can you still open in paraview?

**Comment by singima** [comment]
@tjhei Yes, I just texted it now and I can run my xdmf modified version of step-40 on my machine.

**Comment by tjhei** [comment]
I was trying to run the hdf5 tests on my machine, but they fail on master for me as well:
```
$ ctest -R hdf5
76% tests passed, 8 tests failed out of 34

The following tests FAILED:
        1566 - base/hdf5_03.mpirun=1.debug (Failed)
        1567 - base/hdf5_03.mpirun=1.release (Failed)
        1568 - base/hdf5_03.mpirun=4.debug (Failed)
        1569 - base/hdf5_03.mpirun=4.release (Failed)
        1572 - base/hdf5_05.mpirun=1.debug (Failed)
        1573 - base/hdf5_05.mpirun=1.release (Failed)
        1574 - base/hdf5_05.mpirun=4.debug (Failed)
        1575 - base/hdf5_05.mpirun=4.release (Failed)

```

I will investigate.

**Review by peterrum — APPROVED** [review]

**Comment by tjhei** [comment]
Well, XDMF has nothing to do with the base/hdf5* tests (they test the c++ HDF5 API and not the graphical output). So I think this is safe to merge. I am keeping track of the failing tests at #13638

**Pull request merged**